pkg/proxy/proxy/transparent.go

```diff
@@ -175,6 +203,55 @@ func (t *TransparentProxyService) ServeMetadata(addr string) {
 	}
 }
 
+// Check that the requested url is allowed by the network policy.
+func (proxy TransparentProxyService) CheckNetworkPolicy(req *http.Request, ctx *goproxy.ProxyCtx) (*http.Request, *http.Response) {
+	if proxy.Mode != EnforcementMode {
+		return req, nil
+	}
+	url := req.URL
+	if proxy.Policy != nil {
+		for _, rule := range proxy.Policy.Rules {
+			if url.Hostname() != rule.Host {
+				continue
+			}
+			if urlSatisfiesRule(url, rule) {
+				return req, nil
+			}
+		}
+	}
+	log.Printf("Request to %s blocked by network policy", url.String())
+	errorMessage := fmt.Sprintf("Access to %s is blocked by the proxy's network policy", url.String())
+	return req, goproxy.NewResponse(req, goproxy.ContentTypeText, http.StatusForbidden, errorMessage)
+}
+
+type MatchingStrategy int
+
+const (
+	PathPrefix MatchingStrategy = iota
+	FullPath
+)
+
+type NetworkPolicy struct {
```

Comment:
I feel like this stuff should actually go in a separate package or, at the very least, file. Either one is fine with me but something like policy.Rule or netpolicy.Rule would probably be cleaner than long names here.